model/pod/pod.go

```diff
@@ -196,6 +196,33 @@ func (o TaskContainerCreationOptions) IsZero() bool {
 	return o.Image == "" && o.MemoryMB == 0 && o.CPU == 0 && o.OS == "" && o.Arch == "" && len(o.EnvVars) == 0 && len(o.EnvSecrets) == 0
 }
 
+// Secret is a sensitive secret that a pod can access. The secret is managed
+// in an integrated secrets storage service.
+type Secret struct {
+	// Name is the friendly name of the secret.
+	Name string `bson:"name,omitempty" json:"name,omitempty" yaml:"name,omitempty"`
+	// ExternalID is the unique external resource identifier for a secret that
+	// already exists in the secrets storage service.
+	ExternalID string `bson:"external_id,omitempty" json:"external_id,omitempty" yaml:"external_id,omitempty"`
```

Comment:
Should there be a validation on `ExternalID` only existing when `Owned` is `false`? It seems like it would be invalid to own the secret and also have an external ID.

Reply:
Maybe, but I want to keep it flexible by not assuming that this is how secrets will be shared yet (if we could make this assumption, I would get rid of `Owned` entirely and just decide whether the secret is owned based on the presence/absence of `ExternalID`). There's also no good place to validate it right now since the only way to create a pod is through a temporary REST route which will go away once pods are used in the task lifecycle.